src/Runtime/Distributions/DiscreteChar.cs

```diff
@@ -245,6 +242,13 @@ public static DiscreteChar Create(double probabilityOutsideRanges, IEnumerable<C
             return new DiscreteChar(builder.GetResult());
         }
 
+        /// <summary>
+        /// Creates an unitialized distribution which can't be used until assigned
+        /// </summary>
+        /// <returns>The created distribution.</returns>
+        [Construction(UseWhen = "IsNull", Priority = 2)]
```

Comment:
The only reason you would need Priority is if IsNull overlapped with another UseWhen predicate.  Which one does it overlap with?

Reply:
When `DiscreteChar` `IsNull`, then trying to use any other properties (as you mentioned previously) will trigger `NullReferenceException`. So if quoting checks for cosntructors in the wrong order (lets say for `IsDigit` first) then stuff will crash.

But, actually all Is* properties are not used in hot loops, so I will probably will make them return false on null objects. In this case priority won't be needed and I will revert the `ConstructionAttribute` changes

Reply:
This change is gone